### PR TITLE
CSP-842 Creates MyApprenticeship record

### PR DESCRIPTION
**Pull request by DevenKShah**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeAan.Application.UnitTests/Services/ApprenticeServiceTests/GetStagedApprenticeTests.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        Response<StagedApprentice?> stagedApprenticeResponse,
+        CancellationToken cancellationToken)
+    {
+        stagedApprenticeResponse.ResponseMessage.StatusCode = HttpStatusCode.OK;
+        clientMock.Setup(c => c.GetStagedApprentice(lastName, dateOfBirth.ToString("yyyy-MM-dd"), email, cancellationToken)).ReturnsAsync(stagedApprenticeResponse);
+
+        var result = await sut.GetStagedApprentice(lastName, dateOfBirth, email, cancellationToken);
+
+        result.Should().NotBeNull();
+    }
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Application.UnitTests/Services/ApprenticeServiceTests/GetStagedApprenticeTests.cs:45]: add a check for at least some of the data, lastname would be sufficient

File: src/SFA.DAS.ApprenticeAan.Application.UnitTests/Services/MyApprenticeshipServiceTests.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        MyApprenticeship myApprenticeship,
+        string lastName,
+        string email,
+        DateTime dateOfBirth,
+        CancellationToken cancellationToken)
+    {
+        Mock<IOuterApiClient> clientMock = new();
+
+        Response<MyApprenticeship> myApprenticeshipNotFoundResponse = new(string.Empty, new(HttpStatusCode.NotFound), () => null!);
+        Response<MyApprenticeship> myApprenticeshiOkResponse = new(string.Empty, new(HttpStatusCode.OK), () => myApprenticeship);
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Application.UnitTests/Services/MyApprenticeshipServiceTests.cs:52]: typo p missing in myApprenticeshiOkResponse

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerPostTests.cs [hunk 3]
```diff
@@ -1,95 +1,133 @@
...
+        sessionModel.ApprenticeDetails.Name.Should().Be(user.FullName());
+        sessionModel.MyApprenticeship.Should().Be(myApprenticeship);
+    }
+
+    private OnboardingSessionModel GetSessionModel(bool hasSeenPreview = true)
+    {
+        OnboardingSessionModel model = new() { HasSeenPreview = hasSeenPreview };
        model.ProfileData.Add(new ProfileModel { Id = ProfileDataId.HasPreviousEngagement, Value = null });
        return model;
    }
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerPostTests.cs:132]: this line seems to have some weird characters

**Review by MarkFCain — APPROVED** [review]

File: src/SFA.DAS.ApprenticeAan.Application.UnitTests/Services/MyApprenticeshipServiceTests.cs (hunk 2 above)
Comment (DevenKShah) [reply src/SFA.DAS.ApprenticeAan.Application.UnitTests/Services/MyApprenticeshipServiceTests.cs:52]: fixed

File: src/SFA.DAS.ApprenticeAan.Application.UnitTests/Services/ApprenticeServiceTests/GetStagedApprenticeTests.cs (hunk 1 above)
Comment (DevenKShah) [reply src/SFA.DAS.ApprenticeAan.Application.UnitTests/Services/ApprenticeServiceTests/GetStagedApprenticeTests.cs:45]: It is a mocked object, so it does not make sense to check any properties of mocked object.

**Review by MarkFCain — DISMISSED** [review]

**Review by MarkFCain — APPROVED** [review]

**Pull request merged**